pkg/vulnsrc/vulnsrc.go

```diff
@@ -137,60 +137,86 @@ type Optimizer interface {
 }
 
 func (o fullOptimizer) Optimize() error {
-	err := o.dbc.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
-		severity, title, description, references := vulnerability.GetDetail(cveID)
-		vuln := types.Vulnerability{
-			Title:       title,
-			Description: description,
-			Severity:    severity.String(),
-			References:  references,
-		}
-		if err := o.dbc.PutVulnerability(tx, cveID, vuln); err != nil {
-			return xerrors.Errorf("failed to put vulnerability: %w", err)
-		}
-		return nil
+	err := o.dbOp.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
+		return o.fullOptimize(cveID, tx)
 	})
 	if err != nil {
 		return xerrors.Errorf("failed to iterate severity: %w", err)
 	}
 
-	if err := o.dbc.DeleteSeverityBucket(); err != nil {
+	if err := o.dbOp.DeleteSeverityBucket(); err != nil {
 		return xerrors.Errorf("failed to delete severity bucket: %w", err)
 	}
 
-	if err := o.dbc.DeleteVulnerabilityDetailBucket(); err != nil {
+	if err := o.dbOp.DeleteVulnerabilityDetailBucket(); err != nil {
 		return xerrors.Errorf("failed to delete vulnerability detail bucket: %w", err)
 	}
 
 	return nil
 
 }
 
+// TODO: Until we can dependency inject, we need to monkey patch sadly
+var (
+	getDetailFunc = vulnerability.GetDetail
+)
+
+func (o fullOptimizer) fullOptimize(cveID string, tx *bolt.Tx) error {
+	severity, vs, title, description, references := getDetailFunc(cveID)
+	vuln := types.Vulnerability{
+		Title:          title,
+		Description:    description,
+		Severity:       severity.String(),
+		References:     references,
+		VendorSeverity: vs,
+	}
+
+	if err := o.dbOp.PutVulnerability(tx, cveID, vuln); err != nil {
+		return xerrors.Errorf("failed to put vulnerability: %w", err)
+	}
+	return nil
+}
+
 type lightOptimizer struct {
-	dbc db.Operation
+	dbConfig db.VulnOperation
+	dbOp     db.Operation
 }
 
 func (o lightOptimizer) Optimize() error {
-	err := o.dbc.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
-		// get correct severity
-		sev, _, _, _ := vulnerability.GetDetail(cveID)
-
-		// overwrite unknown severity with correct severity
-		if err := o.dbc.PutSeverity(tx, cveID, sev); err != nil {
-			return xerrors.Errorf("failed to put severity: %w", err)
-		}
-		return nil
+	err := o.dbOp.ForEachSeverity(func(tx *bolt.Tx, cveID string, _ types.Severity) error {
+		return o.lightOptimize(cveID, tx)
 	})
 	if err != nil {
 		return xerrors.Errorf("failed to iterate severity: %w", err)
 	}
 
-	if err = o.dbc.DeleteVulnerabilityDetailBucket(); err != nil {
+	if err = o.dbOp.DeleteVulnerabilityDetailBucket(); err != nil {
 		return xerrors.Errorf("failed to delete vulnerability detail bucket: %w", err)
 	}
 	return nil
 }
+
+func (o lightOptimizer) lightOptimize(cveID string, tx *bolt.Tx) error {
+	// get correct severity
+	severity, vendorSeverity, _, _, _ := getDetailFunc(cveID)
+	vuln := types.Vulnerability{
+		Severity:       severity.String(),
```

Comment:
Is this necessary? We have `Severity` in the full database for backward compatibility because an old trivy client reads the severity from `Severity` key, but it doesn't look necessary in the light database.

Reply:
It's not but I'm keeping this to make logic flow easier when it comes to using it. Mixing sources by using information from Full DB when it's a request for Light DB and vice versa isn't a good pattern IMO.

Reply:
Now, a new client no longer needs `Severity` in both light and full DB, right? Would you explain why sources are mixed a bit more?

Reply:
Ah – I got your point. Sorry I misunderstood earlier. I will remove this.

Reply:
Removed https://github.com/aquasecurity/trivy-db/pull/32/commits/c47c05449be94fe880be11e46c0f6aed18631e26